pkg/operator/target_config_reconciler_v311_00.go

```diff
@@ -151,6 +162,34 @@ func managePod_v311_00_to_latest(client coreclientv1.ConfigMapsGetter, recorder
 	return resourceapply.ApplyConfigMap(client, recorder, configMap)
 }
 
+func checkForFeatureGates(featureGateLister configlistersv1.FeaturesLister) (map[string]bool, error) {
+	featureGateListConfig, err := featureGateLister.Get("cluster")
+	var featureGates = make(map[string]bool)
+	if apierrors.IsNotFound(err) {
+		glog.Infof("features.config.openshift.io/cluster: not found, so return empty feature gates")
+		return nil, err
+	}
+	var enabledFeatureSets, disabledFeatureSets []string
+	currentFeatureSetConfig := featureGateListConfig.Spec.FeatureSet
```

Comment:
I think it's a little cleaner passing in the featureSet into the generateFeatureGates function